Add Gordon Peak Module Reference Board (x86/Apollo Lake)

gnuless opened the pull request:
Final refactorization of the Apollo Lake SoC (vs the Up Squared, its first board target) followed by
board description and documentation of Intel's Gordon Peak automotive dev board.

ulfalizer commented on boards/x86/gpmrb/doc/index.rst:
```diff
@@ -0,0 +1,133 @@
+   be a file :file:`zephyr.strip` which is the same binary with the
+   debugging information removed. Because this binary is being burned
+   into a limited firmware flash area, we will use the smaller,
+   stripped version.
```
Maybe this could be turned into part of the main text instead of being a note. Think some people were trying to get rid of some `note`s earlier.

gnuless replied:
Done.

ulfalizer commented on boards/x86/gpmrb/doc/index.rst:
```diff
@@ -0,0 +1,133 @@
+   into a limited firmware flash area, we will use the smaller,
+   stripped version.
+
+And move the resulting application to your home directory for use in the
```
Could drop the "and" maybe.

gnuless replied:
Done!

ulfalizer commented on boards/x86/gpmrb/doc/index.rst:
```diff
@@ -0,0 +1,133 @@
+
+   $ python Platform/ApollolakeBoardPkg/Script/StitchLoader.py -i ~/LEAFHILD.X64.0070.R01.1805070352.bin -s Outputs/apl/Stitch_Components.zip -o ~/sbl.bin
+
+And now the file :file:`sbl.bin` in your home directory contains a firmware
```
Could drop the "and" maybe.

gnuless replied:
Yep, I could and I did! :)

ulfalizer commented on boards/x86/gpmrb/Kconfig.board:
```diff
@@ -0,0 +1,9 @@
+#
+
+config BOARD_GPMRB
+	bool "Gordon Peak MRB"
```
Seems a bit weird that these symbols have a prompt. I wonder if going into the `menuconfig` and changing the board there was ever expected to work. Might not if other settings for the board are set by the `-D<BOARD>`.

Unrelated though, so could keep it for consistency. :)

gnuless replied:
Yeah.. for the moment I'm just following suit, rather than trying to set a standard.

ulfalizer reviewed (commented):
Kconfig parts look okay. Don't know anything about this board though.

Commit:
boards/x86/gpmrb: add Gordon Peak MRB board

Add board support (and documentation) for the Intel Gordon Peak
Module Reference Board, a dev board based on the Apollo Lake SoC.

Signed-off-by: Charles E. Youse <charles.youse@intel.com>

gnuless commented on boards/x86/gpmrb/doc/index.rst:
```diff
@@ -0,0 +1,132 @@
+
+The Intel Gordon Peak Module Reference Board (GP MRB) is used in
+the automotive industry for the development of in-vehicle applications
+like heads-up displays and entertainment systems.
```
fixed

gnuless commented on boards/x86/gpmrb/doc/index.rst:
```diff
@@ -0,0 +1,132 @@
+Build a Zephyr application; for instance, to build the ``hello_world``
+application for the GP MRB:
+
+   .. zephyr-app-commands::
```
fixed

gnuless commented on boards/x86/gpmrb/doc/index.rst:
```diff
@@ -0,0 +1,132 @@
+
+There will now be two files named :file:`LEAFHILD.X64.0070.D01.1805070344.bin`
+and :file:`LEAFHILD.X64.0070.R01.1805070352.bin` or similar in your home
+directory, which are the debug and release versions of the binary packages,
```
changed

gnuless commented on boards/x86/gpmrb/doc/index.rst:
```diff
@@ -0,0 +1,132 @@
+
+.. code-block:: console
+
+   $ python Platform/ApollolakeBoardPkg/Script/StitchLoader.py -i ~/LEAFHILD.X64.0070.R01.1805070352.bin -s Outputs/apl/Stitch_Components.zip -o ~/sbl.bin
```
changed

gnuless commented on boards/x86/gpmrb/doc/index.rst:
```diff
@@ -0,0 +1,132 @@
+   $ python Platform/ApollolakeBoardPkg/Script/StitchLoader.py -i ~/LEAFHILD.X64.0070.R01.1805070352.bin -s Outputs/apl/Stitch_Components.zip -o ~/sbl.bin
+
+Now the file :file:`sbl.bin` in your home directory contains a firmware
+image with SBL and the Zephyr application, ready to flash to the GP MRB!
```
fixed

gnuless commented on boards/x86/gpmrb/doc/index.rst:
```diff
@@ -0,0 +1,132 @@
+Flash the Image
+===============
+
+Connect the IOC to the GP MRB and connect the USB cable to your development
```
If they have a GP MRB, they should know what it is.

gnuless commented on boards/x86/gpmrb/doc/index.rst:
```diff
@@ -0,0 +1,132 @@
+================================
+
+The Slim Bootloader (see the next step) requires binary firmware images
+specific to our target platform: in this instance, the "Leaf Hill" firmware.
```
fixed

gnuless commented on boards/x86/gpmrb/doc/index.rst:
```diff
@@ -0,0 +1,132 @@
+.. note::
+
+   Refer to the instructions with the IOC and/or GP MRB for further
+   information on flasing the firmware.
```
fixed

gnuless commented on boards/x86/gpmrb/doc/index.rst:
```diff
@@ -0,0 +1,132 @@
+Launch Zephyr
+=============
+
+Connect to UART 2 on the GP MRB and press the "ignition" button. After
```
It's actually a button labelled "ignition".

gnuless commented on soc/x86/apollo_lake/doc/supported_features.rst:
```diff
@@ -0,0 +1,57 @@
+
+   .. code-block:: console
+
+    / {
```
Yep.

gnuless commented on boards/x86/gpmrb/doc/index.rst:
```diff
@@ -0,0 +1,132 @@
+      :goals: build
+
+
+This will result in an output file named :file:`zephyr.elf`, which
```
fixed

gnuless commented on boards/x86/gpmrb/doc/index.rst:
```diff
@@ -0,0 +1,132 @@
+===============
+
+Connect the IOC to the GP MRB and connect the USB cable to your development
+machine. Then, using the Intel Platform Flash tools supplied with your
```
What would you recommend that I put here?  I'm documenting how to use something, not trying to sell it to anyone :-)

gnuless commented on boards/x86/gpmrb/doc/index.rst:
```diff
@@ -0,0 +1,132 @@
+
+.. note::
+
+   Refer to the instructions with the IOC and/or GP MRB for further
```
I assume so. There's a ton of documentation available to people who get these things, right?

dbkinder commented on soc/x86/apollo_lake/doc/supported_features.rst:
```diff
@@ -0,0 +1,57 @@
+Zephyr what the resulting master clock is.
+
+Use devicetree to set the value of the ``PRV_CLOCK_PARAMS`` register in
+the UART block of interest. Typically an overlay ``up_squared.overlay``
```
Should this say, ``mrb.overlay`` or more generically, ``boardname.overlay``.  Currently this file is only included into an MRB-specific document, so it's odd to see up_squared mentioned.
